eng/source-build-release.yml

```diff
@@ -6,198 +6,236 @@ pool:
+  steps:
+  - checkout: self
+  - checkout: installer
+
+  - template: templates/steps/get-build-info.yml
+    parameters:
+      dotnetStagingPipelineResource: dotnet-staging-pipeline-resource
+      getAssociatedPipelineRuns: false
+
+  - template: templates/steps/validate-tag.yml
+    parameters:
+      useCustomTag: ${{ parameters.useCustomTag }}
+      customTag: ${{ parameters.customTag }}
+      repoPath: $(Build.SourcesDirectory)/installer
+
+  - script: |
+      set -euxo pipefail
+
+      query='query { repository(owner: "${{ variables.announcementOrg }}", name: "${{ variables.announcementRepo }}") { id } }'
+      echo "${query}"
+      repo_id=$( gh api graphql -f query="$query" --template '{{.data.repository.id}}' )
+      echo ${{ variables.announcementOrg }}/${{ variables.announcementRepo }} repo ID is ${repo_id}
+
+      query='query { repository(name: "${{ variables.announcementRepo }}", owner: "${{ variables.announcementOrg }}") { discussionCategories(first: 10) { edges { node { id, name } } } } }'
+      echo "${query}"
+      category_id=$( gh api graphql -f query="$query" --template '{{range .data.repository.discussionCategories.edges}}{{if eq .node.name "Announcements"}}{{.node.id}}{{end}}{{end}}' )
+      echo Discussion Category ID is ${category_id}
+
+      echo "##vso[task.setvariable variable=RepoId;]${repo_id}"
+      echo "##vso[task.setvariable variable=DiscussionCategoryId;]${category_id}"
+    displayName: Get category IDs
+    env:
+      GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+
+  - script: |
+      set -euxo pipefail
+      echo Repo ID is $(RepoId)
+      echo Discussion Category ID is $(DiscussionCategoryId)
+
+      # Set environment variables that go in the announcement template
+      export TAG=$(Tag)
+      export RUNTIME_VERSION="$(RuntimeVersion)"
+      export RELEASE_CHANNEL="$(ReleaseChannel)"
+      export SDK_VERSION="$(SdkVersion)"
+      export TAG_URL="https://github.com/dotnet/installer/releases/tag/$TAG"
+      export RELEASE_NOTES_URL="https://github.com/dotnet/core/blob/main/release-notes/$RELEASE_CHANNEL/$RUNTIME_VERSION/$SDK_VERSION.md"
+      export RELEASE_DATE=$(date +"%B %Y") # e.g. "March 2022"
+
+      template_path="$(Build.SourcesDirectory)/$(Build.Repository.Name)/eng"
+      template="$(envsubst < ${template_path}/source-build-release-announcement.md)"
+      # Get the line in the template that is prefixed with "Title:" and remove the prefix
+      title=$(echo "$template" | grep "^Title:" | cut -d " " -f2-)
+      # Get the inverse of the above selection
+      body=$(echo "$template" | grep -v "^Title:")
+
+      query='mutation($RepoId: ID!, $categoryId: ID!, $body: String!, $title: String!) { createDiscussion(input: {repositoryId: $RepoId, categoryId: $categoryId, body: $body, title: $title}) { discussion { url } } }'
+
+      if [ ${{ parameters.testingMode }} = True ]; then
+        echo "Testing mode is set to True, not submitting announcement."
```

Comment:
For diagnostic purposes, I think it would be useful to emit the announcement content as pipeline output.